### PR TITLE
Change the signature of "handle"

### PR DESCRIPTION
**Pull request by rnjtranjan**

### CODE REVIEW

**Comment by rnjtranjan** [comment]
Fix:https://github.com/composewell/streamly/issues/2485

File: core/src/Streamly/Internal/Data/Stream/StreamD/Exception.hs [hunk 1]
```diff
@@ -83,9 +83,10 @@ gbracket_ bef aft onExc ftry action =
                 Skip s -> return $ Skip (GBracketNormal (Stream step1 s) v)
                 Stop -> aft v >> return Stop
             -- XXX Do not handle async exceptions, just rethrow them.
-            Left e ->
+            Left e -> do
+                strm <- onExc v e (UnStream step1 st)
                 return
-                    $ Skip (GBracketException (onExc v e (UnStream step1 st)))
+                    $ Skip (GBracketException strm)
```
Comment (harendra-kumar) [inline core/src/Streamly/Internal/Data/Stream/StreamD/Exception.hs:89]: Collapse the two lines in a single line.

File: core/src/Streamly/Internal/Data/Stream/StreamD/Exception.hs [hunk 2]
```diff
@@ -301,7 +302,7 @@ bracketUnsafe bef aft =
     gbracket_
         bef
         aft
-        (\a (e :: SomeException) _ -> nilM (aft a >> MC.throwM e))
+        (\a (e :: SomeException) _ -> return $ nilM (aft a >> MC.throwM e))
```
Comment (harendra-kumar) [inline core/src/Streamly/Internal/Data/Stream/StreamD/Exception.hs:305]: Same as previous comment.

**Review by harendra-kumar — CHANGES_REQUESTED** [review]

**Review by harendra-kumar — CHANGES_REQUESTED** [review]
write a changelog entry for breaking change.

**Review by harendra-kumar — APPROVED** [review]

**Pull request merged**